### ISSUE TITLE
Build error: gradle unable to pull .pom /.jar file from mavenCenter

### ISSUE THREAD

**Issue by bbl-jcastaneda** [sarriaroman/photoviewer#239]
It seems as if the .pom and .jar files have been removed from all popular repositories. anybody knows from where I can pull it now? The error I get is below.

Searched in the following locations:
       - https://dl.google.com/dl/android/maven2/com/commit451/PhotoView/1.2.4/PhotoView-1.2.4.pom
       - https://repo.maven.apache.org/maven2/com/commit451/PhotoView/1.2.4/PhotoView-1.2.4.pom
       - file:/builds/BelizeBankLimited/mobilebanking/android/capacitor-cordova-android-plugins/src/main/libs/PhotoView-1.2.4.jar
       - file:/builds/BelizeBankLimited/mobilebanking/android/capacitor-cordova-android-plugins/src/main/libs/PhotoView.jar
       - file:/builds/BelizeBankLimited/mobilebanking/android/app/libs/PhotoView-1.2.4.jar
       - file:/builds/BelizeBankLimited/mobilebanking/android/app/libs/PhotoView.jar
       - https://jcenter.bintray.com/com/commit451/PhotoView/1.2.4/PhotoView-1.2.4.pom
       - file:/builds/BelizeBankLimited/mobilebanking/android/app/src/main/aar/PhotoView-1.2.4.jar
       - file:/builds/BelizeBankLimited/mobilebanking/android/app/src/main/aar/PhotoView.jar

**Comment by tamsel12** [comment]
Yes , We have the same problem from yesterday, We want to move our build to stores today itself. So Please let us know any solutions

**Comment by acasellasnecsia** [comment]
@tamsel12 @bbl-jcastaneda 

I tried to configure Gradle to look for and use a local .aar file instead of downloading it from a repository and it worked for me.

Place the PhotoView-1.2.4.aar file in android/app/libs (create new folder).

Then, modify the build.gradle file (XXX-photoviewer.gradle) of the module:

repositories {
    jcenter()
    **flatDir {
        dirs 'libs'
    }**
}

dependencies {
    **implementation(name: 'PhotoView-1.2.4', ext: 'aar')**
    implementation 'com.squareup.picasso:picasso:2.71828'
}

Sync with Gradle and compile. It worked for me!

Hope this helps!

**Comment by bbl-jcastaneda** [comment]
@acasellasnecsia 

Hi, I also had that as an alternative, but I was unable to locate the .aar file. Would you be able to share how you got that file or how I can generate it by chance?

Thanks!

**Comment by acasellasnecsia** [comment]
Here is the link: https://repo1.maven.org/maven2/com/github/chrisbanes/photoview/library/1.2.4/library-1.2.4.aar.

Hope this helps @bbl-jcastaneda @tamsel12

**Comment by tamsel12** [comment]
Yes, After upgrading this we can able to build the application, but photo is not opening. photoviewer.show not working @acasellasnecsia

**Comment by acasellasnecsia** [comment]
@tamsel12  It's not quite like that, you need to make several calls for the photo to open. I finally gave up on using the plugin and started doing it manually.

**Comment by tamsel12** [comment]
Thiis update works fine in capacitor. But in cordova , photo is not getting opened.

**Comment by lmarcelocc** [comment]
I'm using cordova-android@12.0.1 and with the following patch I was able to compile again and everything is working as before.
[com-sarriaroman-photoviewer+1.3.0.patch](https://github.com/user-attachments/files/16764282/com-sarriaroman-photoviewer%2B1.3.0.patch)

**Comment by Esom** [comment]
> repositories {
> jcenter()
> **flatDir { dirs 'libs' }**
> }

Hi 

> lmarcelocc

You saved me , thanks

**Comment by almaz1213** [comment]
you can download the lib here https://repo.gradle.org/ui/native/remote-repos/com/commit451/PhotoView/1.2.4/
or:
````
repositories{
  jcenter()
  maven {
    url = uri("https://repo.gradle.org/artifactory/remote-repos/")
  }
}
````

**Comment by shamilds** [comment]
> you can download the lib here https://repo.gradle.org/ui/native/remote-repos/com/commit451/PhotoView/1.2.4/ or:
> 
> ```
> repositories{
>   jcenter()
>   maven {
>     url = uri("https://repo.gradle.org/artifactory/remote-repos/")
>   }
> }
> ```

This worked for me

**Comment by rolinger** [comment]
@almaz1213 , @lmarcelocc - I am having an odd issue, hoping either of you can help.

when trying to add this repository in two different environments I am getting different results:

```
repositories{
  jcenter()
  maven {
    url = uri("https://repo.gradle.org/artifactory/remote-repos/")
  }
}
```

I re-installed my entire older environment on a new system and when installing this photoviewer plugin, your above solution worked - simply added the maven repository to the photoviewer.gradle file.

Older environment:
   * cordova v11.1
   * android v11.0
   * nodejs 14.21.3
   * npm 6.14.4, 
   * project compiles with: gradle 7.4.2

Then on my original dev system I upgraded the above older entire environment to some of the latest updates and when it came to this plugin the same solution does not work.  I keep getting the `could not find com.comit451:PhotoView:1.2.4` error.  I added the above fix to `/project/node-modules/com-sarriaroman-photoviewer/src/android/PhotoViewer.gradle` and to `/project/plugins/com-sarriaroman-photoviewer/src/android/Photoviewer.gradle`

Upgraded environment:
   * cordova v12.0
   * android v13.0
   * nodejs v22.16.0
   * npm 11.4.1
   * project compiles with: gradle 8.7

I then added the above repository to my project root `repositories.gradle` file.  But no matter what I do when I build my project I keep getting the same `com.comit451:PhotoView:1.2.4` error.   I believe it should install if I can just get the build to reference the above repository, but for some reason its not recognizing it in any of the gradle files I have added it to.

**Comment by almaz1213** [comment]
@rolinger is https://repo.gradle.org/artifactory/remote-repos/ available from your pc?

**Comment by rolinger** [comment]
@almaz1213 - yes it is, I get an index list of `remote-repos`

```
com/
me/
net/
org/
```

**Comment by rolinger** [comment]
Here is the full error:

```
Starting a Gradle Daemon (subsequent builds will be faster)

> Configure project :app
Adding classpath: com.google.gms:google-services:4.3.15
strippedNativeLibsDir is deprecated and no longer used. Please remove it from your build configuration.

> Task :app:checkDebugAarMetadata FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:checkDebugAarMetadata'.
> Could not resolve all files for configuration ':app:debugRuntimeClasspath'.
   > Could not find com.commit451:PhotoView:1.2.4.
     Searched in the following locations:
       - https://dl.google.com/dl/android/maven2/com/commit451/PhotoView/1.2.4/PhotoView-1.2.4.pom
       - https://repo.maven.apache.org/maven2/com/commit451/PhotoView/1.2.4/PhotoView-1.2.4.pom
       - https://maven.google.com/com/commit451/PhotoView/1.2.4/PhotoView-1.2.4.pom
       - https://jcenter.bintray.com/com/commit451/PhotoView/1.2.4/PhotoView-1.2.4.pom
     Required by:
         project :app
```

**Comment by almaz1213** [comment]
@rolinger 
maven {
    url = uri("https://repo.gradle.org/artifactory/remote-repos/")
  }
seems to be not involved by gradle. check about `maven` instruction for your current gradle version or tray download the lib manually to your (project) libs directory

**Comment by rolinger** [comment]
@almaz1213 thanks.  will do.  Don't have much time tomorrow but I will look into it, def by Monday, report back then.  Thanks.

**Comment by rolinger** [comment]
@almaz1213 - just got back to this issue after the weekend.  Can you clarify your follow up comment....`seems to be not involved by gradle`.   I guess I don't understand what you mean, because its the identical fix that you posted back on April 7th (above) - that fixed worked on one system using gradel 7.4.2, but not on another using a newer gradle 8.7.  Trying to avoid downloading it directly.

**Comment by rolinger** [comment]
@almaz1213 - OK....I figured it out!  Can't believe I did this.

There are 5 different gradle files, only one of them allowed the repository to work.  In my working project I added it to the one correct gradle file not realizing the plugin had created three separate directory structures within the project.  I think i just lucked out on the correct one the first time.  And then for some reason I got on a different brain wavelength on the other computer and started adding the repository to the other two (incorrect) gradle files, neither of them working, before then trying to add them to the 2 /project/android repositories.gradle files (incorrect again).

Add the following repository to your gradle file, just make certain its the CORRECT gradle file:

```
repositories{
  jcenter()
  maven {
    url = uri("https://repo.gradle.org/artifactory/remote-repos/")
  }
}

```
`/project/node_modules/com-sarriaroman-photoviewer/src/android/photoviewer.gradle` - _wrong file_
`/project/plugins/com-sarriaroman-photoviewer/src/android/photoviewer.gradle` - _wrong file_
`/project/platforms/android/repositories.gradle` - _wrong file_
`/project/platforms/android/CordovaLib/repositories.gralde` - _wrong file_

`/project/platforms/android/com-sarriaroman-photoviewer/appname-photoviewer.gradle` - **CORRECT** file
